src/melody/inputs.py

```diff
@@ -1,93 +1,74 @@
 ''' '''
 
-class Fixed(object):
+class Base(object):
```

Comment:
I don't think 'Base' is a good name for a class :-) How about InputBase or just Input?